atomics/T1123/src/T1123.sh

```diff
@@ -0,0 +1,27 @@
+echo "Starting recording, make some noise for $2 seconds!"
+osascript -e '
+    on run argv
+        set theFilePath to POSIX path of item 1 of argv
+        set durn to item 2 of argv
+        tell application "Quicktime Player"
+            start (new audio recording)
+            repeat durn times
+                log (durn)
+                delay 1
+                set durn to (durn - 1)
+            end repeat
+            tell document "Audio Recording"
+                pause
+                save it in POSIX file theFilePath
+                stop
+                close
+            end tell
+            close
+        end tell
+    end run
+' "$1" $2
+RECSIZE=`cat $1 | wc -c`
+if [ $RECSIZE -gt 100000 ]; then
+    echo "Recording complete"
+else echo "Failed" && exit 1
+fi
```

Comment:
```suggestion
# check for non-empty data file
if [ -s "${OUTPATH}" ] ; then
   echo "audio data present" && exit 0
fi
echo "Failed"
exit 1
```

Reply:
unfortunately, even when it doesn't work, it can still create an approx 62KB file, hence the file size check.

Reply:
i've adapted your suggestions into [53e5352](https://github.com/redcanaryco/atomic-red-team/pull/2081/commits/53e53525a89788ab292d8987fc091b0f1a950b94)

Reply:
well then you will have to add a check to make sure that the file exists first.  Otherwise `cat` will cause a `no such file` error if TCC not granted.
```
$ sh T1123.yaml blah.m4a 2
Starting recording, make some noise for 2 seconds!
171:190: execution error: Not authorized to send Apple events to QuickTime Player. (-1743)
cat: blah.m4a: No such file or directory
Failed
```